docs/index.bs

```diff
@@ -2973,23 +2975,21 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
   </section>
 
   <section algorithm>
-    <h3 id="run-service-worker-algorithm"><dfn>Run Service Worker</dfn></h3>
+    <h3 id="partial-setup-algorithm"><dfn>Partial Setup</dfn></h3>
```

Comment:
I feel `Partial Setup` is so not intuitive, how about `Setup Service Worker Global Scope`?

Reply:
Naming this operation is tricky.

If I understand correctly, this operation does not represent how things are actually implemented. In the implementation, you will probably not spin up a whole new process (agent), create a ServiceWorkerGlobalScope JavaScript/V8 object, etc. You will just do the minimum amount of work necessary to ensure that the relevant implementation objects for doing a COEP check are ready.

Whereas in the spec, this operation does "most of" the work of starting up the service worker: basically, everything except running the script.

However, this mismatch is OK, because the observable consequences are the same. Nobody could write a test that distinguishes the spec you have written, from some other version of the spec which only does minimal work.

From this perspective, I think @sisidovski's naming suggestion is more accurate on the spec level. But it is less accurate in terms of the implementation level. I think we should use the spec-accurate name, but add a note explaining the situation. Here is a first draft of such a note; you can check if it is accurate and tweak as desired.

> This algorithm does the minimal setup for the service worker that is necessary to create something usable for security checks like CSP and COEP. This specification ensures that this algorithm is called before any such checks are performed.
>
> In specifications, such security checks require creating a `ServiceWorkerGlobalScope`, a relevant settings object, a realm, and an agent. In implementations, the amount of work required might be much less. So, implementations could do less work in their equivalent of this algorithm, and more work in Run Service Worker, as long as the results are observably equivalent. (And in particular, as long as all security checks have the same result.)

Reply:
> I feel Partial Setup is so not intuitive, how about Setup Service Worker Global Scope?

Sure.  I renamed this.

> If I understand correctly, this operation does not represent how things are actually implemented. In the implementation, you will probably not spin up a whole new process (agent), create a ServiceWorkerGlobalScope JavaScript/V8 object, etc. You will just do the minimum amount of work necessary to ensure that the relevant implementation objects for doing a COEP check are ready.

Yes.  My plan is executing a part of [EmbeddedWorkerInstance::Start](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/service_worker/embedded_worker_instance.cc;l=230;drc=c0f955cad0bf2a21ec83e95e369eb050c030013a), but not run another process or not fully setup ServiceWorkerGlobalScope.

I also add notes to tell that.  I am not sure how to make them pretty printed, though.